src/hsmm_mvpy/models.py

```diff
@@ -807,7 +807,7 @@ def EM(self, magnitudes, parameters, maximization=True, magnitudes_to_fix=None,
         else:
             lkh_prev = -np.inf
             while i < max_iteration :#Expectation-Maximization algorithm
-                if i > min_iteration and tolerance > (lkh-lkh_prev)/np.abs(lkh_prev):
+                if i >= min_iteration and (np.isneginf(lkh) or tolerance > (lkh-lkh_prev)/np.abs(lkh_prev)):
```

Comment:
Why the np.isneginf(lkh)? I didn't have this situation before I think

Reply:
This only happened in a very specific situation (feeding in already optimal mags/params, still going through EM), but I don't think it hurts. It might not be necessary with the i >= min_iteration, but I wasn't entirely sure.

Reply:
Yes it can stay, I was just curious